### PR TITLE
Update to Mirage 3

### PR DESCRIPTION
**Pull request by gaborigloi**
* Update `lib/block.ml` to match the new [`Mirage_types_lwt.BLOCK`](http://docs.mirage.io/mirage-types-lwt/Mirage_types_lwt/module-type-BLOCK/index.html) interface
* Move from the polymorphic variant result type to the standard `result` type, and use the `rresult` result combinator library

### CODE REVIEW

File: opam [hunk 1]
```diff
@@ -14,11 +14,14 @@ depends: [
   "ocamlfind"
   "lwt" {>= "2.4.3"}
   "cstruct" {>= "1.9"}
+  "mirage-block"
   "mirage-types-lwt" {>= "3.0.0"}
   "ipaddr"
   "io-page"
```
Comment (mseri) [inline opam]: You also need "io-page-unix"

**Comment by gaborigloi** [comment]
Should we do the jbuilder port in this PR or a subsequent one?

**Comment by mseri** [comment]
Fair enough. Let’s hear what the maintainers prefer

**Comment by mseri** [comment]
Ping @yomimono

File: opam (hunk 1 above)
Comment (yomimono) [reply opam]: Hm, why `-unix`?

**Comment by yomimono** [comment]
I don't really know much about this package, but the port to Mirage 3 looks reasonable to me. I think the real experts on this repository are @djs55 and @avsm , but if they don't have time to review in the next couple of days I'm happy to merge.

Jbuilder port in a separate PR would be great :)

File: opam (hunk 1 above)
Comment (gaborigloi) [reply opam]: Hmm, it seems that `Executable parse_test` in the `_oasis` file requires `io-page.unix`. In the jbuilder port in a followup PR we can make this a `{test}` dependency.

Comment (yomimono) [reply opam]: Ah, I see.  As long as it's marked `{test}` it's fine.

Comment (mseri) [reply opam]: Oh I see, without additional changes to the `_oasis` configuration we cannot currently flag it as test. Imho is fine to leave it as is in this PR, and make it `{test}` in the `jbuilder` port. What do you think?

**Comment by mseri** [comment]
🎉 Thanks! Time for the `jbuilder` port now :)

**Comment by djs55** [comment]
@yomimono thanks for the review and merge! Sorry I didn't respond to the ping :(

**Comment by gaborigloi** [comment]
We've ported the lib to jbuilder with @mseri , PR is https://github.com/mirage/ocaml-vhd/pull/46

**Pull request merged**